core/src/com/unciv/models/ModConstants.kt

```diff
@@ -71,32 +83,37 @@ class ModConstants {
-        if (other.cityStrengthFromTechsFullMultiplier != defaults.cityStrengthFromTechsFullMultiplier) cityStrengthFromTechsFullMultiplier = other.cityStrengthFromTechsFullMultiplier
-        if (other.cityStrengthFromGarrison != defaults.cityStrengthFromGarrison) cityStrengthFromGarrison = other.cityStrengthFromGarrison
-        if (other.unitSupplyPerPopulation != defaults.unitSupplyPerPopulation) unitSupplyPerPopulation = other.unitSupplyPerPopulation
-        if (other.minimalCityDistance != defaults.minimalCityDistance) minimalCityDistance = other.minimalCityDistance
-        if (other.minimalCityDistanceOnDifferentContinents != defaults.minimalCityDistanceOnDifferentContinents) minimalCityDistanceOnDifferentContinents = other.minimalCityDistanceOnDifferentContinents
-        if (other.unitUpgradeCost != defaults.unitUpgradeCost) unitUpgradeCost = other.unitUpgradeCost
-        if (other.naturalWonderCountMultiplier != defaults.naturalWonderCountMultiplier) naturalWonderCountMultiplier = other.naturalWonderCountMultiplier
-        if (other.naturalWonderCountAddedConstant != defaults.naturalWonderCountAddedConstant) naturalWonderCountAddedConstant = other.naturalWonderCountAddedConstant
-        if (other.ancientRuinCountMultiplier != defaults.ancientRuinCountMultiplier) ancientRuinCountMultiplier = other.ancientRuinCountMultiplier
-        if (other.spawnIceBelowTemperature != defaults.spawnIceBelowTemperature) spawnIceBelowTemperature = other.spawnIceBelowTemperature
-        if (other.maxLakeSize != defaults.maxLakeSize) maxLakeSize = other.maxLakeSize
-        if (other.riverCountMultiplier != defaults.riverCountMultiplier) riverCountMultiplier = other.riverCountMultiplier
-        if (other.minRiverLength != defaults.minRiverLength) minRiverLength = other.minRiverLength
-        if (other.maxRiverLength != defaults.maxRiverLength) maxRiverLength = other.maxRiverLength
-        if (other.religionLimitBase != defaults.religionLimitBase) religionLimitBase = other.religionLimitBase
-        if (other.religionLimitMultiplier != defaults.religionLimitMultiplier) religionLimitMultiplier = other.religionLimitMultiplier
-        if (other.pantheonBase != defaults.pantheonBase) pantheonBase = other.pantheonBase
-        if (other.pantheonGrowth != defaults.pantheonGrowth) pantheonGrowth = other.pantheonGrowth
+        for (field in this::class.java.declaredFields) {
+            val value = field.get(other)
+            if (field.get(defaults).equals(value)) continue
+            field.set(this, value)
+        }
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other !is ModConstants) return false
+        return equalsReflected(other)
+    }
+
+    override fun hashCode(): Int {
+        var result = 0
+        for (field in this::class.java.declaredFields) {
+            result = result * 31 + field.get(this).hashCode()
+        }
+        return result
+    }
```

Comment:
I see this is the same as the Vector2 hash implementation, I wonder if there's a builtin "multihash" function that accepts an iterable of hashes and returns one...